[CASE test_no_output]
class A:
    pass
class B:
    pass
[out]

[CASE mypy_output_as_comments]
reveal_type(1)  # E: Revealed type is 'builtins.int'
reveal_type('one')  # E: Revealed type is 'builtins.str'
[out]

[CASE mypy_output_as_text]
reveal_type(1)
reveal_type('one')
[out]
main:1: error: Revealed type is 'builtins.int'
main:2: error: Revealed type is 'builtins.str'
